FaunaDB.Client/Types/Attributes.cs

```diff
@@ -50,10 +50,35 @@ public class FaunaFieldAttribute : Attribute
         /// </summary>
         public object DefaultValue { get; set; }
 
+        /// <summary>
+        /// Overrides the fauna type <see cref="FaunaDB.Types"/> namespace for
+        /// list of accepted values.
+        /// Type should inherit from <see cref="FaunaDB.Types.Value"/>.
+        /// Such as, <see cref="TimeV"/>, <see cref="DateV"/>, <see cref="DoubleV"/>, etc
+        ///
+        /// Not fully supprted, only use for the <see cref="Encoder"/> specifically to enforce
+        /// DateV or TimeV for <see cref="DateTime"/> and <see cref="DateTimeOffset"/> properties
+        /// </summary>
+        public Type Type { get; }
+
         public FaunaFieldAttribute(string name)
         {
             Name = name;
         }
+        public FaunaFieldAttribute(string name, Type type)
```

Comment:
Are those combination of constructor really necessary? I wonder if we add a new field to this attribute how exponential this will be. I think this kinda reinforce the my argument above.